src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3043,15 +2936,8 @@ bool c_typecheck_baset::gcc_vector_types_compatible(
 
 void c_typecheck_baset::typecheck_expr_binary_arithmetic(exprt &expr)
 {
-  if(expr.operands().size()!=2)
-  {
-    error().source_location = expr.source_location();
-    error() << "operator '" << expr.id() << "' expects two operands" << eom;
-    throw 0;
-  }
-
-  exprt &op0=expr.op0();
-  exprt &op1=expr.op1();
+  exprt &op0 = to_binary_expr(expr).op0();
+  exprt &op1 = to_binary_expr(expr).op1();
```

Comment:
I think we should do `to_binary_expr` just once.

Reply:
done